### PR TITLE
Backup: test release process for Backup plugin prototype

### PR DESCRIPTION
**Pull request by samiff**
#### Changes proposed in this Pull Request:

* Testing out the Jetpack monorepo tools/scripts to ensure they work for releasing other plugins such as the Backup prototype.
* Testing 0.1.0-beta release for `/projects/plugins/backup`. Will need to merge this PR to master, then cut the beta release branch.

#### Jetpack product discussion

* Internal: p1HpG7-b9F-p2

#### Does this pull request change what data or activity we track or use?

* No.

#### Testing instructions:

* `tools/changelogger-release.sh -b plugins/backup` has been run, review updated changelogs for accuracy.
* Other change made was adding `release-branch-prefix` for the Backup plugin.

### CODE REVIEW

**Comment by anomiex** [comment]
I noticed that it didn't have a change like https://github.com/Automattic/jetpack/pull/19930/files#diff-e97d4ccadff4cc0c54af458896780d0f7e2ccfcfd2b9dcb1f11570a46d8d3b82R95. Which is because it doesn't set autoloader-suffix at all, which is because it doesn't have stuff in .gitattributes like https://github.com/Automattic/jetpack/blob/a83ce23b8dfb6bb43d8afe41bf1baf3f26051f7b/projects/plugins/jetpack/.gitattributes#L10-L15 

Should it?

**Comment by samiff** [comment]
@anomiex I've updated gitattributes in 15841f8 but I'm not so sure on how the `autoloader-suffix` is set up. I tried running `changelogger-release.sh` again after changing the gitattributes and adding a blank autoloader-suffix, but didn't see that it updated the suffix at all.

File: projects/plugins/backup/.gitattributes [hunk 1]
```diff
@@ -1,13 +1,24 @@
-# Files not needed to be distributed in the package.
-.gitattributes    export-ignore
-.github/          export-ignore
-package.json      export-ignore
+# Files to include in mirror repo: Automattic/jetpack-backup-plugin
+build/**                      production-include
+jetpack-backup.php            production-include
+src/**                        production-include
```
Comment (anomiex) [inline projects/plugins/backup/.gitattributes:4]: These line should be unnecessary. `production-include` is basically for overriding `.gitignore` but these aren't ignored.

```suggestion
```

File: projects/plugins/backup/.gitattributes [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+babel.config.js               production-exclude
+build/**/*.js.map             production-exclude
+changelog/**                  production-exclude
+composer.lock                 production-exclude
+gulpfile.babel.js             production-exclude
+package.json                  production-exclude
+phpunit.xml.dist              production-exclude
+readme.md                     production-exclude
+tests/**                      production-exclude
+webpack.config.js             production-exclude
```
Comment (anomiex) [inline projects/plugins/backup/.gitattributes:24]: Optional, but we don't really need changelogger or jetpack-autoloader (the package) at runtime.
```suggestion
/vendor/automattic/jetpack-autoloader/**                 production-exclude
/vendor/automattic/jetpack-changelogger/**               production-exclude
webpack.config.js             production-exclude
```

**Review by anomiex — COMMENTED** [review]
I see that CI is giving the intended help message about `autoloader-suffix`. Let me know if you think we can improve that somehow.

I already know that it says to add "...ⓥversion" but really wants "...ⓥ0_1_0_alpha" (i.e. the actual version number), but I'm not sure whether the script generating the suggestion can reliably determine the actual version number.

Also, I see the build is failing. It looks like that's because composer used symlinks for all the local packages, but the mirroring doesn't like that. Jetpack avoids this by having its production build step delete `vendor/` then run `COMPOSER_MIRROR_PATH_REPOS=1 composer install -o --no-dev --classmap-authoritative --prefer-dist` to get a "production" version of `vendor/`. Backup should probably do that too.

**Comment by samiff** [comment]
> I see that CI is giving the intended help message about autoloader-suffix

Thanks @anomiex  Must have missed that but got it added in now 👍 

I've adjusted the build scripts as well, think this is ready to try for the mirrored beta release now?

**Review by anomiex — APPROVED** [review]
Looks sane, and more importantly the build artifact for mirroring looks sane too.

Before the actual release someone should install the plugin from the mirror repo version to confirm it works properly.

**Pull request merged**